specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,689 @@
+        HRESULT add_CustomItemSelected(
+            [in] ICoreWebView2StagingCustomItemSelectedEventHandler* eventHandler,
+            [out] EventRegistrationToken* token);
+
+        /// Remove an event handler previously added with `add_CustomItemSelected`.
+        HRESULT remove_CustomItemSelected(
+            [in] EventRegistrationToken token);
+    }
+
+    /// Represents a collection of `ContextMenuItem` objects. Used to get, remove and add
+    /// `ContextMenuItem` objects at the specified index.
+    [uuid(f562a2f5-c415-45cf-b909-d4b7c1e276d3), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItemCollection : IUnknown
+    {
+        /// Gets the number of `ContextMenuItem` objects contained in the `ContextMenuItemCollection`.
+        [propget] HRESULT Count([out, retval] UINT32* value);
+
+        /// Gets the `ContextMenuItem` at the specified index.
+        HRESULT GetValueAtIndex([in] UINT32 index,
+            [out, retval] ICoreWebView2ContextMenuItem** value);
+
+        /// Removes the `ContextMenuItem` at the specified index.
+        HRESULT RemoveValueAtIndex([in] UINT32 index);
+
+        /// Inserts the `ContextMenuItem` at the specified index.
+        HRESULT InsertValueAtIndex([in] UINT32 index, [in] ICoreWebView2ContextMenuItem* value);
+    }
+
+    [uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+    interface ICoreWebView2_4 : ICoreWebView2_3
+    {
+        /// Add an event handler for the `ContextMenuRequested` event.
+        /// `ContextMenuRequested` event is raised when a context menu is requested by the user
+        /// and the content inside WebView hasn't disabled context menus.
+        /// The host has the option to create their own context menu with the information provided in 
+        /// the event or can add items to or remove items from WebView context menu.
+        /// If the host doesn't handle the event, WebView will display the default context menu.
+        HRESULT add_ContextMenuRequested(
+            [in] ICoreWebView2StagingContextMenuRequestedEventHandler* eventHandler,
+            [out] EventRegistrationToken* token);
+
+        /// Remove an event handler previously added with `add_ContextMenuRequested`.
+        HRESULT remove_ContextMenuRequested(
+            [in] EventRegistrationToken token);
+    }
+
+    /// A continuation of the ICoreWebView2Environment interface.
+    [uuid(04d4fe1d-ab87-42fb-a898-da241d35b63c), object, pointer_default(unique)]
+    interface ICoreWebView2Environment5 : ICoreWebView2Environment4
+    {
+        /// Create a `ContextMenuItem` object to insert into the WebView context menu.
+        /// CoreWebView2 will rewind the stream before decoding. Command Id for new 
+        /// custom menu items will be unique for the lifespan of the ContextMenuRequested
+        /// event. They will be a unique value between 52600 and 52650.
```

Comment:
Probably want a bigger range. Only 50 context menu items seems like it could be cutting it close. Unless there's a technical limitation can we pick a max a few orders of magnitudes larger like 10000 or something?